src/modules/rating/controllers/admin-rating.controller.ts

```diff
@@ -68,26 +68,32 @@ export class AdminRatingController {
 		@Query('status') status: string,
 		@Query('cursor') cursor?: string | null,
 		@Query('size') size: number = 10,
+		@Query('sortOrder') sortOrder: 'asc' | 'desc' = 'desc', // Default to descending order for newest to oldest
 	): Promise<FilteredByStatusResponseDTO> {
 		try {
 			let ratings: Rating[]
-
-			if (cursor === '') {
-				ratings = await this.ratingRepo.getByStatus(status, null, size)
+			if (!cursor) {
+				// If cursor is not provided, fetch the first page
+				ratings = await this.ratingRepo.getByStatus(
+					status,
+					null, // Start from the beginning
+					size,
+					sortOrder,
+				)
 			} else {
+				// If cursor is provided, fetch the next page
 				ratings = await this.ratingRepo.getByStatus(
 					status,
 					cursor,
 					size,
+					sortOrder,
 				)
 			}
-
 			let newCursor: string | null = null
-
 			if (ratings.length > 0) {
 				if (ratings.length > size) {
-					newCursor = ratings[size]._id
-					ratings.splice(size)
+					newCursor = ratings[size - 1]._id.toString() // Use the ID of the last rating in the current page
```

Comment:
Logic in this line is updated. Is the old version a bug ?

Reply:
the old version not really bug just not suit the requirement so i change to set it stay choosing last rating of first page and when customer click next page (new cusor) the id been choosing will keep at top and still load the data from next page

Reply:
If you choose last rating of the current page (let's say ID is **10**), the first rating of next page will be **10** as well, which is not the desired behavior.